libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/security/Encryptor.java

```diff
@@ -284,17 +284,7 @@ public static String encryptWithRSA(PublicKey publicKey, String data) {
      * @return Encrypted data.
      */
     public static byte[] encryptWithRSABytes(PublicKey publicKey, String data) {
-        if (publicKey == null || TextUtils.isEmpty(data)) {
-            return null;
-        }
-        try {
-            final Cipher cipher = Cipher.getInstance(RSA_PKCS1);
-            cipher.init(Cipher.ENCRYPT_MODE, publicKey);
-            return cipher.doFinal(data.getBytes());
-        } catch (Exception e) {
-            SalesforceAnalyticsLogger.e(null, TAG, "Error during asymmetric encryption using RSA", e);
-        }
-        return null;
+        return encryptWithPublicKey(publicKey, data, RSA_PKCS1);
```

Comment:
Doesn't the key know its own algorithm? https://developer.android.com/reference/java/security/Key.html#getAlgorithm()

Reply:
That method call would return "RSA", not "RSA/ECB/PKCS1Padding".

Reply:
Maybe we could our own key object wrapped around Android's and Encryptor would consume that.

Reply:
Maybe we could, but I don't think it would add too much value, not at this time anyway. We still need this API the way it is because S1 uses this for push notification encryption/decryption. There's not much we would add to `PublicKey` or `PrivateKey` right now. I would prefer leaving this as is for now. Perhaps we can re-evaluate if there is more extensive usage of keys.